lib/docker.rb

```diff
@@ -119,6 +119,29 @@ def ping(connection = self.connection)
     connection.get('/_ping')
   end
 
+  # Determine if the server is podman or docker.
+  def podman?(connection = self.connection)
+    @attrs ||= {}
+    return @attrs[:podman] if @attrs[:podman]
+
+    @attrs[:podman] = !(
+      Array(version(connection)['Components']).find do |component|
+        component['Name'] && component['Name'].include?('Podman')
+      end
+    ).nil?
+
+    @attrs[:podman]
+  end
+
+  # Determine if the session is rootless.
+  def rootless?(connection = self.connection)
```

Comment:
Since this isn't used programmatically for other API calls, we can keep this one uncached.

Reply:
I've ended up needing to use it quite a bit elsewhere so it makes sense to cache it to me.